lib/chef/knife/core/ui.rb

```diff
@@ -66,7 +66,7 @@ def highline
       # the logger API.
       def msg(message)
         begin
-          stdout.puts message
+          stdout.puts message unless (message.respond_to?('empty?') && message.empty?)
```

Comment:
I'm wondering if, instead of changing this method, we should change the `output` method below.  `output` is what *should* be called by all core knife plugin when printing an object while .msg is for more manual output control.

While you've changed all the call sites of `msg` in the core plugins, this change will still change the formatting of other plugins.

I agree that we should never print a single blank line, but the current change would change the output of a plugin such as the following:

```
class Foo < Chef::Knife
  def run
    ui.msg({})
  end
end
```